test/python/synthesis/test_synthesis.py

```diff
@@ -149,6 +149,16 @@ def assertDebugOnly(self):  # FIXME: when at python 3.10+ replace with assertNoL
             )
             self.assertIn("Requested fidelity:", record.getMessage())
 
+    def assertPickle(self, decomp: TwoQubitControlledUDecomposer):
```

Comment:
There isn't a test using this method so this code is never executed. If you want to use this method you'll have to add a test that calls it. Something prefixed with `test_` will do that.

Reply:
I updated the test, and now it's failing.
with a standard gate it fails with: `TypeError: cannot pickle 'qiskit._accelerate.circuit.StandardGate' object`
with a customized gate it fails with: `AttributeError: Can't pickle local object 'TestTwoQubitControlledUDecompose.test_assertPickle.<locals>.CustomXXGate'
`